contrib/vyper.asciidoc

```diff
@@ -47,6 +47,13 @@ The human readable ABI code (in JSON format) can be obtained by then running the
 
 vyper -f json ~/hello_world.v.py
 
+=== Reading and writing data
+Ethereum smart contracts can write data to two places. Ethereum's global state trie or Ethereum's chain data. While it is costly to store, read and modify data, these storage operations are a necessary component of most smart contracts. 
 
+==== Global state
+The state variables in a given Ethereum smart contract are stored in Ethereum's global state trie. A given smart contract can only store, read and modify data specifically in relation to that contracts address i.e. smart contracts can not read or write to other smart contracts. TODO continue with more information about this in relation to Vyper
```

Comment:
You are missing a ', and I would put the ie in parentheses.
contract's address (i.e. smart contracts ... )